### ISSUE TITLE
Optimized setting based on system configuration

### ISSUE THREAD

**Issue by gbazad93** [ivadomed/ivadomed#920]
While working with ivadomed and almost any other library, I always needed to choose the best parameters for the training process and better generalization. Such a setting might be hard to get by the end-user, especially if they have less expertise in deep learning (which is expected in the medical domain). It would be great if Ivadomed includes an automatic parameter setting (learning rate, batch-size, model depth, et al.) based on the system configuration (for example, GPU memory, RAM etc) and user metadata (some predefined information about the task, dataset, etc.). 

Example references: 
[nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation](https://www.nature.com/articles/s41592-020-01008-z)

**Comment by charleygros** [comment]
That's excellent feedback, thanks for that @gbazad93 

Our approach for now was to setup some generic config file for various task, eg for [classification](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/config/config_classification.json).

But you are so right, it depends on much more elements than the task only.

The article you have shared is highly relevant and interesting.

If you would like to participate to its implementation in ivadomed, please feel very welcome to.
Otherwise, no worries, we may suggest it as an internship project at Neuropoly? Let's see.

**Comment by gbazad93** [comment]
@charleygros I am always keen to a challenging work and do love to learn from you more. Possibly after finishing my current research (by the end of October) I can send you a request :).
